bokeh-demo/bokeh_demo/data_ingestion.py

```diff
@@ -40,13 +38,22 @@ def fit(self, X: pd.DataFrame, y=None) -> CategoryColumnConverter:
     def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         X = X.copy()
         for col in self.category_cols:
-            X[col] = X[col].apply(lambda x: x.value)
+            X[col] = X[col].apply(self.try_convert_enum)
         X[self.category_cols] = X[self.category_cols].astype("str")
         return X
 
+    @staticmethod
+    def try_convert_enum(x):
+        """Converts enum to its value if it has one, otherwise returns the value itself."""
+        try:
+            return x.value
+        except AttributeError:
+            return x
+
 
 class CreatePersonSource(BaseEstimator, TransformerMixin):
-    """Takes in the exams dataframe and returns another dataframe for plotting Lexis plots."""
+    """Takes in the exams dataframe and returns another dataframe (with one row per person)
+    for plotting Lexis plots."""
 
     person_results_map: dict[int, list[int]] = {}
     person_exam_ages_map: dict[int, list[int]] = {}
```

Comment:
Declare and describe this. Neat idea to simply use a lookup table!

```suggestion
    person_exam_ages_map: dict[int, list[int]] = {}
    person_exam_inds_map: dict[int, list[int]]
    """Mapping from PID to list of that persons exams' indices"""
```